// @ant-design/pro-layout/lib/components/AppsLogoComponents/DefaultContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DefaultContent = void 0;

var _jsxRuntime = require("react/jsx-runtime");

var _react = require("react");

var _emotion = require("../../emotion");

var _ProLayoutContext = require("../../ProLayoutContext");

var _css = require("./css");

var _index = require("./index");

var DefaultContent = function DefaultContent(props) {
  var designToken = (0, _react.useContext)(_ProLayoutContext.ProLayoutContext);
  var appList = props.appList,
      _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'ant' : _props$prefixCls;
  return (0, _jsxRuntime.jsx)("div", {
    className: "".concat(prefixCls, "-basicLayout-apps-content"),
    children: (0, _jsxRuntime.jsx)("ul", {
      className: (0, _emotion.cx)("".concat(prefixCls, "-basicLayout-apps-content-list"), _css.appContentListCss),
      children: appList === null || appList === void 0 ? void 0 : appList.map(function (app, index) {
        return (0, _jsxRuntime.jsx)("li", {
          className: (0, _emotion.cx)("".concat(prefixCls, "-basicLayout-apps-content-list-item"), (0, _css.getAppContentLisItem)(designToken)),
          children: (0, _jsxRuntime.jsxs)("a", {
            href: app.url,
            target: app.target,
            rel: "noreferrer",
            children: [(0, _index.defaultRenderLogo)(app.icon), (0, _jsxRuntime.jsxs)("div", {
              children: [(0, _jsxRuntime.jsx)("div", {
                children: app.title
              }), app.desc ? (0, _jsxRuntime.jsx)("span", {
                children: app.desc
              }) : null]
            })]
          })
        }, index);
      })
    })
  });
};

exports.DefaultContent = DefaultContent;
