// @ant-design/pro-layout/lib/components/AppsLogoComponents/SimpleContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.renderLogo = exports.SimpleContent = void 0;

var _jsxRuntime = require("react/jsx-runtime");

var _react = require("react");

var _emotion = require("../../emotion");

var _ProLayoutContext = require("../../ProLayoutContext");

var _simple = require("./simple");

/**
 * simple模式渲染logo的方式
 *
 * @param logo
 * @param title
 * @returns
 */
var renderLogo = function renderLogo(logo, title) {
  if (logo && typeof logo === 'string') {
    return (0, _jsxRuntime.jsx)("img", {
      src: logo,
      alt: "logo"
    });
  }

  if (typeof logo === 'function') {
    return logo();
  }

  if (!logo && title && typeof title === 'string') {
    var symbol = title.substring(0, 1);
    return (0, _jsxRuntime.jsx)("div", {
      id: "avatarLogo",
      children: symbol
    });
  }

  return logo;
};

exports.renderLogo = renderLogo;

var SimpleContent = function SimpleContent(props) {
  var designToken = (0, _react.useContext)(_ProLayoutContext.ProLayoutContext);
  var appList = props.appList,
      _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'ant' : _props$prefixCls;
  return (0, _jsxRuntime.jsx)("div", {
    className: "".concat(prefixCls, "-basicLayout-apps-content"),
    children: (0, _jsxRuntime.jsx)("ul", {
      className: (0, _emotion.cx)("".concat(prefixCls, "-basicLayout-apps-content-list"), _simple.appContentListCss),
      children: appList === null || appList === void 0 ? void 0 : appList.map(function (app, index) {
        return (0, _jsxRuntime.jsx)("li", {
          className: (0, _emotion.cx)("".concat(prefixCls, "-basicLayout-apps-content-list-item"), (0, _simple.getAppContentLisItem)(designToken)),
          children: (0, _jsxRuntime.jsxs)("a", {
            href: app.url,
            target: app.target,
            rel: "noreferrer",
            children: [renderLogo(app.icon, app.title), (0, _jsxRuntime.jsx)("div", {
              children: (0, _jsxRuntime.jsx)("div", {
                children: app.title
              })
            })]
          })
        }, index);
      })
    })
  });
};

exports.SimpleContent = SimpleContent;
