// @ant-design/pro-layout/lib/components/FooterToolbar/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FooterToolbar = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

require("antd/lib/config-provider/style");

var _configProvider = _interopRequireDefault(require("antd/lib/config-provider"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _jsxRuntime = require("react/jsx-runtime");

var _proUtils = require("@ant-design/pro-utils");

var _omit = _interopRequireDefault(require("omit.js"));

var _react = require("react");

var _reactDom = require("react-dom");

var _emotion = require("../../emotion");

var _index = require("../../index");

var _ProLayoutContext = require("../../ProLayoutContext");

var _templateObject, _templateObject2, _templateObject3;

var _excluded = ["children", "className", "extra", "style", "renderContent"];

var FooterToolbar = function FooterToolbar(props) {
  var designToken = (0, _react.useContext)(_ProLayoutContext.ProLayoutContext);
  var children = props.children,
      className = props.className,
      extra = props.extra,
      style = props.style,
      renderContent = props.renderContent,
      restProps = (0, _objectWithoutProperties2.default)(props, _excluded);

  var _useContext = (0, _react.useContext)(_configProvider.default.ConfigContext),
      getPrefixCls = _useContext.getPrefixCls;

  var prefixCls = props.prefixCls || getPrefixCls('pro');
  var baseClassName = "".concat(prefixCls, "-footer-bar");
  var value = (0, _react.useContext)(_index.RouteContext);
  var width = (0, _react.useMemo)(function () {
    var hasSiderMenu = value.hasSiderMenu,
        isMobile = value.isMobile,
        siderWidth = value.siderWidth;

    if (!hasSiderMenu) {
      return undefined;
    } // 0 or undefined


    if (!siderWidth) {
      return '100%';
    }

    return isMobile ? '100%' : "calc(100% - ".concat(siderWidth, "px)"); // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [value.collapsed, value.hasSiderMenu, value.isMobile, value.siderWidth]);
  var dom = (0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [(0, _jsxRuntime.jsx)("div", {
      className: (0, _emotion.cx)("".concat(baseClassName, "-left"), (0, _emotion.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2.default)(["\n            flex: 1;\n          "])))),
      children: extra
    }), (0, _jsxRuntime.jsx)("div", {
      className: (0, _emotion.cx)("".concat(baseClassName, "-right"), (0, _emotion.css)(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2.default)(["\n            > * {\n              margin-right: 8px;\n              &:last-child {\n                margin: 0;\n              }\n            }\n          "])))),
      children: children
    })]
  });
  /** 告诉 props 是否存在 footerBar */

  (0, _react.useEffect)(function () {
    if (!value || !(value === null || value === void 0 ? void 0 : value.setHasFooterToolbar)) {
      return function () {};
    }

    value === null || value === void 0 ? void 0 : value.setHasFooterToolbar(true);
    return function () {
      var _value$setHasFooterTo;

      value === null || value === void 0 ? void 0 : (_value$setHasFooterTo = value.setHasFooterToolbar) === null || _value$setHasFooterTo === void 0 ? void 0 : _value$setHasFooterTo.call(value, false);
    }; // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  var renderDom = (0, _jsxRuntime.jsx)("div", (0, _objectSpread2.default)((0, _objectSpread2.default)({
    className: (0, _emotion.cx)(className, baseClassName, (0, _emotion.css)(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2.default)(["\n          position: fixed;\n          right: 0;\n          bottom: 0;\n          z-index: 99;\n          display: flex;\n          align-items: center;\n          width: 100%;\n          padding: 0 24px;\n          line-height: 64px;\n          background-color: ", ";\n          border-top: 1px solid ", ";\n          transition: width 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);\n          background-color: rgba(255, 255, 255, 0.58);\n          -webkit-backdrop-filter: blur(8px);\n          backdrop-filter: blur(8px);\n        "])), designToken.layoutBgColor, designToken.borderColorSplit)),
    style: (0, _objectSpread2.default)({
      width: width
    }, style)
  }, (0, _omit.default)(restProps, ['prefixCls'])), {}, {
    children: renderContent ? renderContent((0, _objectSpread2.default)((0, _objectSpread2.default)((0, _objectSpread2.default)({}, props), value), {}, {
      leftWidth: width
    }), dom) : dom
  }));
  return !(0, _proUtils.isBrowser)() ? renderDom : /*#__PURE__*/(0, _reactDom.createPortal)(renderDom, document.body);
};

exports.FooterToolbar = FooterToolbar;
