// @ant-design/pro-layout/lib/components/SettingDrawer/BlockCheckbox.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BlockCheckbox = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

require("antd/lib/tooltip/style");

var _tooltip = _interopRequireDefault(require("antd/lib/tooltip"));

var _jsxRuntime = require("react/jsx-runtime");

var _icons = require("@ant-design/icons");

var _classnames = _interopRequireDefault(require("classnames"));

var _react = require("react");

var _emotion = require("../../emotion");

var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6, _templateObject7;

var BlockCheckbox = function BlockCheckbox(_ref) {
  var value = _ref.value,
      configType = _ref.configType,
      onChange = _ref.onChange,
      list = _ref.list,
      prefixCls = _ref.prefixCls;
  var baseClassName = "".concat(prefixCls, "-drawer-block-checkbox");
  var dom = (0, _react.useMemo)(function () {
    var domList = (list || []).map(function (item) {
      return (0, _jsxRuntime.jsx)(_tooltip.default, {
        title: item.title,
        children: (0, _jsxRuntime.jsx)("div", {
          className: (0, _emotion.cx)((0, _classnames.default)("".concat(baseClassName, "-item"), "".concat(baseClassName, "-item-").concat(item.key), "".concat(baseClassName, "-").concat(configType, "-item")), (0, _emotion.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2.default)(["\n              position: relative;\n              width: 44px;\n              height: 36px;\n              margin-right: 16px;\n              overflow: hidden;\n              background-color: #f7f8fa;\n              border-radius: 4px;\n              box-shadow: 0 1px 2.5px 0 rgba(0, 0, 0, 0.18);\n              cursor: pointer;\n              &::before {\n                position: absolute;\n                top: 0;\n                left: 0;\n                width: 33%;\n                height: 100%;\n                background-color: #fff;\n                content: '';\n              }\n              &::after {\n                position: absolute;\n                top: 0;\n                left: 0;\n                width: 100%;\n                height: 25%;\n                background-color: #fff;\n                content: '';\n              }\n            "]))), item.key === 'realDark' && (0, _emotion.css)(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2.default)(["\n                background-color: rgba(0, 21, 41, 0.85);\n                &::before {\n                  background-color: rgba(0, 21, 41, 0.65);\n                  content: '';\n                }\n                &::after {\n                  background-color: rgba(0, 21, 41, 0.85);\n                }\n              "]))), item.key === 'light' && (0, _emotion.css)(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2.default)(["\n                &::before {\n                  background-color: #fff;\n                  content: '';\n                }\n                &::after {\n                  background-color: #fff;\n                }\n              "]))), (item.key === 'dark' || item.key === 'side') && (0, _emotion.css)(_templateObject4 || (_templateObject4 = (0, _taggedTemplateLiteral2.default)(["\n                &::before {\n                  z-index: 1;\n                  background-color: #001529;\n                  content: '';\n                }\n                &::after {\n                  background-color: #fff;\n                }\n              "]))), item.key === 'top' && (0, _emotion.css)(_templateObject5 || (_templateObject5 = (0, _taggedTemplateLiteral2.default)(["\n                &::before {\n                  background-color: transparent;\n                  content: '';\n                }\n                &::after {\n                  background-color: #001529;\n                }\n              "]))), item.key === 'mix' && (0, _emotion.css)(_templateObject6 || (_templateObject6 = (0, _taggedTemplateLiteral2.default)(["\n                ::before {\n                  background-color: #fff;\n                  content: '';\n                }\n                &::after {\n                  background-color: #001529;\n                }\n              "])))),
          onClick: function onClick() {
            return onChange(item.key);
          },
          children: (0, _jsxRuntime.jsx)(_icons.CheckOutlined, {
            className: (0, _emotion.cx)("".concat(baseClassName, "-selectIcon"), (0, _emotion.css)(_templateObject7 || (_templateObject7 = (0, _taggedTemplateLiteral2.default)(["\n                position: absolute;\n                right: 6px;\n                bottom: 4px;\n                color: var(--ant-primary-color);\n                font-weight: bold;\n                font-size: 14px;\n                pointer-events: none;\n                .action {\n                  color: var(--ant-primary-color);\n                }\n              "])))),
            style: {
              display: value === item.key ? 'block' : 'none'
            }
          })
        })
      }, item.key);
    });
    return domList; // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [value, list === null || list === void 0 ? void 0 : list.length, onChange]);
  return (0, _jsxRuntime.jsx)("div", {
    className: baseClassName,
    style: {
      display: 'flex',
      minHeight: 42
    },
    children: dom
  });
};

exports.BlockCheckbox = BlockCheckbox;
