// @ant-design/pro-layout/lib/components/SettingDrawer/ThemeColor.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ThemeColor = void 0;

require("antd/lib/tooltip/style");

var _tooltip = _interopRequireDefault(require("antd/lib/tooltip"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _jsxRuntime = require("react/jsx-runtime");

var _icons = require("@ant-design/icons");

var _react = _interopRequireDefault(require("react"));

var _emotion = require("../../emotion");

var _templateObject, _templateObject2, _templateObject3;

var _excluded = ["color", "check"];

var Tag = /*#__PURE__*/_react.default.forwardRef(function (_ref, ref) {
  var color = _ref.color,
      check = _ref.check,
      rest = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return (0, _jsxRuntime.jsx)("div", (0, _objectSpread2.default)((0, _objectSpread2.default)({}, rest), {}, {
    style: {
      backgroundColor: color
    },
    ref: ref,
    children: check ? (0, _jsxRuntime.jsx)(_icons.CheckOutlined, {}) : ''
  }));
});

var ThemeColor = function ThemeColor(_ref2) {
  var value = _ref2.value,
      colorList = _ref2.colorList,
      onChange = _ref2.onChange,
      formatMessage = _ref2.formatMessage;

  if (!colorList || (colorList === null || colorList === void 0 ? void 0 : colorList.length) < 1) {
    return null;
  }

  return (0, _jsxRuntime.jsx)("div", {
    className: (0, _emotion.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2.default)(["\n        margin-top: 16px;\n        overflow: hidden;\n      "]))),
    children: (0, _jsxRuntime.jsx)("div", {
      className: (0, _emotion.cx)('theme-color-content', (0, _emotion.css)(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2.default)([""])))),
      children: colorList === null || colorList === void 0 ? void 0 : colorList.map(function (_ref3) {
        var key = _ref3.key,
            color = _ref3.color;
        if (!key) return null;
        return (0, _jsxRuntime.jsx)(_tooltip.default, {
          title: formatMessage({
            id: "app.setting.themecolor.".concat(key)
          }),
          children: (0, _jsxRuntime.jsx)(Tag, {
            className: (0, _emotion.cx)('theme-color-block', (0, _emotion.css)(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2.default)(["\n                    float: left;\n                    width: 20px;\n                    height: 20px;\n                    margin-top: 8px;\n                    margin-right: 8px;\n                    color: #fff;\n                    font-weight: bold;\n                    text-align: center;\n                    border-radius: 2px;\n                    cursor: pointer;\n                  "])))),
            color: color,
            check: value === color,
            onClick: function onClick() {
              return onChange && onChange(color);
            }
          })
        }, color);
      })
    })
  });
};

exports.ThemeColor = ThemeColor;
