// @ant-design/pro-layout/es/components/SiderMenu/SiderMenu.js
import "antd/es/menu/style";
import _Menu from "antd/es/menu";
import "antd/es/avatar/style";
import _Avatar from "antd/es/avatar";
import "antd/es/space/style";
import _Space from "antd/es/space";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import "antd/es/config-provider/style";
import _ConfigProvider from "antd/es/config-provider";
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import "antd/es/layout/style";
import _Layout from "antd/es/layout";
import _taggedTemplateLiteral from "@babel/runtime/helpers/esm/taggedTemplateLiteral";
var _excluded = ["isMobile", "collapsed"];

var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6, _templateObject7, _templateObject8, _templateObject9, _templateObject10, _templateObject11, _templateObject12, _templateObject13, _templateObject14, _templateObject15, _templateObject16, _templateObject17, _templateObject18, _templateObject19, _templateObject20;

import { createElement as _createElement } from "react";
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import classNames from 'classnames';
import { useContext, useEffect, useMemo } from 'react';
import { css, cx, injectGlobal, keyframes } from '../../emotion';
import { ProLayoutContext } from '../../ProLayoutContext';
import { AppsLogoComponents, defaultRenderLogo } from '../AppsLogoComponents';
import { ArrowSvgIcon } from './Arrow';
import { BaseMenu } from './BaseMenu';
import { MenuCounter } from './Counter';
export var defaultIconCss = function defaultIconCss(designToken) {
  return css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n  position: absolute;\n  top: 18px;\n  z-index: 101;\n  width: 24px;\n  height: 24px;\n  font-size: 14px;\n  color: ", ";\n  text-align: center;\n  border-radius: 40px;\n  right: -13px;\n  background-color: ", ";\n  transition: transform 0.3s;\n  font-size: 16px;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  cursor: pointer;\n  .anticon {\n    font-size: 14px;\n  }\n\n  &:hover {\n    color: ", ";\n  }\n"])), designToken.sider.collapsedButtonTextColor, designToken.sider.collapsedButtonBgColor, designToken.sider.collapsedButtonHoverTextColor);
};
export var proLayoutTitleHide = keyframes(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n0% {\n  display: none;\n  width: 1px;\n  margin: 0;\n  overflow: hidden;\n  white-space: nowrap;\n  opacity: 0;\n}\n80% {\n  display: none;\n  width: 1px;\n  margin: 0;\n  overflow: hidden;\n  white-space: nowrap;\n  opacity: 0;\n}\n100% {\n  display: unset;\n  height: auto;\n  opacity: 1;\n}"])));
export var siderCss = css(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n  position: relative;\n  background: transparent;\n\n  &-menu {\n    position: relative;\n    z-index: 10;\n    min-height: 100%;\n  }\n"])));
export var siderTitleViewCss = function siderTitleViewCss(designToken) {
  return css(_templateObject4 || (_templateObject4 = _taggedTemplateLiteral(["\n  position: relative;\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  padding: 16px 16px;\n  color: ", ";\n  border-bottom: 1px solid ", ";\n  cursor: pointer;\n\n  > a {\n    display: flex;\n    align-items: center;\n    justify-content: center;\n    min-height: 22px;\n    font-size: 22px;\n  }\n\n  img {\n    display: inline-block;\n    height: 22px;\n    vertical-align: middle;\n  }\n\n  h1 {\n    display: inline-block;\n    height: 22px;\n    margin: 0 0 0 6px;\n    color: ", ";\n    font-weight: 600;\n    font-size: 16px;\n    line-height: 22px;\n    vertical-align: middle;\n    animation: ", " 0.35s;\n  }\n"])), designToken.sider.menuTextColor, designToken.sider.menuItemDividerColor, designToken.sider.menuTitleTextColor, proLayoutTitleHide);
};
var Sider = _Layout.Sider;

var CollapsedIcon = function CollapsedIcon(props) {
  var isMobile = props.isMobile,
      collapsed = props.collapsed,
      rest = _objectWithoutProperties(props, _excluded);

  var designToken = useContext(ProLayoutContext);
  if (isMobile && collapsed) return null;
  return _jsx("div", _objectSpread(_objectSpread({}, rest), {}, {
    className: cx(props.className, defaultIconCss(designToken), css("\n          box-shadow: 0 2px 8px -2px rgba(0,0,0,0.05), 0 1px 4px -1px rgba(25,15,15,0.07), 0 0 1px 0 rgba(0,0,0,0.08);\n          &:hover {\n            box-shadow: 0 4px 16px -4px rgba(0,0,0,0.05), 0 2px 8px -2px rgba(25,15,15,0.07), 0 1px 2px 0 rgba(0,0,0,0.08);\n          }\n        "), !isMobile && css("\n        > svg {\n          transform: ".concat(!collapsed ? 'rotate(90deg)' : 'rotate(-90deg)', ";\n        }\n          ")), isMobile && css("  > svg {\n            right: -16px;\n            top: 72px;\n            transform: ".concat(collapsed ? 'rotate(-90deg) translate(8px, 0px)' : 'rotate(90deg)', ";\n          "))),
    children: _jsx(ArrowSvgIcon, {})
  }));
};
/**
 * 渲染 title 和 logo
 *
 * @param props
 * @param renderKey
 * @returns
 */


export var renderLogoAndTitle = function renderLogoAndTitle(props) {
  var renderKey = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'menuHeaderRender';
  var logo = props.logo,
      title = props.title,
      layout = props.layout;
  var renderFunction = props[renderKey || ''];

  if (renderFunction === false) {
    return null;
  }

  var logoDom = defaultRenderLogo(logo);

  var titleDom = _jsx("h1", {
    children: title !== null && title !== void 0 ? title : 'Ant Design Pro'
  });

  if (renderFunction) {
    // when collapsed, no render title
    return renderFunction(logoDom, props.collapsed ? null : titleDom, props);
  }
  /**
   * 收起来时候直接不显示
   */


  if (props.isMobile) {
    return null;
  }

  if (layout === 'mix' && renderKey === 'menuHeaderRender') return false;

  if (props.collapsed) {
    return _jsx("a", {
      children: logoDom
    }, "title");
  }

  return _jsxs("a", {
    children: [logoDom, titleDom]
  }, "title");
};

var SiderMenu = function SiderMenu(props) {
  var _classNames, _props$menu2;

  var collapsed = props.collapsed,
      originCollapsed = props.originCollapsed,
      fixSiderbar = props.fixSiderbar,
      menuFooterRender = props.menuFooterRender,
      _onCollapse = props.onCollapse,
      theme = props.theme,
      siderWidth = props.siderWidth,
      isMobile = props.isMobile,
      onMenuHeaderClick = props.onMenuHeaderClick,
      _props$breakpoint = props.breakpoint,
      breakpoint = _props$breakpoint === void 0 ? 'lg' : _props$breakpoint,
      style = props.style,
      layout = props.layout,
      _props$menuExtraRende = props.menuExtraRender,
      menuExtraRender = _props$menuExtraRende === void 0 ? false : _props$menuExtraRende,
      links = props.links,
      menuContentRender = props.menuContentRender,
      collapsedButtonRender = props.collapsedButtonRender,
      prefixCls = props.prefixCls,
      avatarProps = props.avatarProps,
      rightContentRender = props.rightContentRender,
      actionsRender = props.actionsRender,
      onOpenChange = props.onOpenChange,
      headerHeight = props.headerHeight,
      logoStyle = props.logoStyle;
  var showSiderExtraDom = useMemo(function () {
    if (isMobile) return false;
    if (layout === 'mix') return false;
    return true;
  }, [isMobile, layout]);
  var designToken = useContext(ProLayoutContext);
  var context = useContext(_ConfigProvider.ConfigContext);
  var baseClassName = "".concat(prefixCls, "-sider");

  var _MenuCounter$useConta = MenuCounter.useContainer(),
      flatMenuKeys = _MenuCounter$useConta.flatMenuKeys;

  var siderClassName = classNames("".concat(baseClassName), (_classNames = {}, _defineProperty(_classNames, "".concat(baseClassName, "-fixed"), fixSiderbar), _defineProperty(_classNames, "".concat(baseClassName, "-layout-").concat(layout), layout && !isMobile), _defineProperty(_classNames, "".concat(baseClassName, "-light"), theme !== 'dark'), _classNames));
  var headerDom = renderLogoAndTitle(props);
  var extraDom = menuExtraRender && menuExtraRender(props);
  var menuDom = useMemo(function () {
    return menuContentRender !== false && flatMenuKeys && /*#__PURE__*/_createElement(BaseMenu, _objectSpread(_objectSpread({}, props), {}, {
      key: "base-menu",
      mode: "inline",
      handleOpenChange: onOpenChange,
      style: {
        width: '100%'
      },
      className: "".concat(baseClassName, "-menu")
    }));
  }, [baseClassName, flatMenuKeys, menuContentRender, onOpenChange, props]);
  var linksMenuItems = (links || []).map(function (node, index) {
    return {
      className: "".concat(baseClassName, "-link"),
      label: node,
      key: index
    };
  });
  var menuRenderDom = useMemo(function () {
    return menuContentRender ? menuContentRender(props, menuDom) : menuDom;
  }, [menuContentRender, menuDom, props]);
  var avatarDom = useMemo(function () {
    return avatarProps && _jsxs(_Space, {
      align: "center",
      className: cx("".concat(baseClassName, "-actions-avatar"), css(_templateObject5 || (_templateObject5 = _taggedTemplateLiteral(["\n              font-size: 14px;\n              padding: 8px;\n              border-radius: ", ";\n              & > * {\n                cursor: pointer;\n              }\n              &:hover {\n                background: rgba(0, 0, 0, 0.018);\n              }\n            "])), designToken.borderRadiusBase)),
      children: [_jsx(_Avatar, _objectSpread({}, avatarProps)), avatarProps.title && !collapsed && _jsx("span", {
        children: avatarProps.title
      })]
    });
  }, [avatarProps, baseClassName, collapsed, designToken.borderRadiusBase]);
  var actionsDom = useMemo(function () {
    if (!actionsRender) return null;
    return _jsx(_Space, {
      align: "center",
      size: 4,
      direction: collapsed ? 'vertical' : 'horizontal',
      className: cx(["".concat(baseClassName, "-actions-list"), css(_templateObject6 || (_templateObject6 = _taggedTemplateLiteral(["\n              color: ", ";\n              animation: ", " 0.3s;\n            "])), designToken.sider.menuTextColorSecondary, proLayoutTitleHide), collapsed && css(_templateObject7 || (_templateObject7 = _taggedTemplateLiteral(["\n                margin-bottom: 8px;\n                animation: none;\n              "])))]),
      children: actionsRender === null || actionsRender === void 0 ? void 0 : actionsRender(props).map(function (item, index) {
        return _jsx("div", {
          className: css(_templateObject8 || (_templateObject8 = _taggedTemplateLiteral(["\n                  padding: 6px;\n                  line-height: 16px;\n                  font-size: 16px;\n                  cursor: pointer;\n                  border-radius: ", ";\n                  &:hover {\n                    background: rgba(0, 0, 0, 0.018);\n                  }\n                "])), designToken.borderRadiusBase),
          children: item
        }, index);
      })
    });
  }, // eslint-disable-next-line react-hooks/exhaustive-deps
  [actionsRender, baseClassName, collapsed]);
  var appsDom = useMemo(function () {
    return _jsx(AppsLogoComponents, {
      appList: props.appList,
      prefixCls: props.prefixCls
    });
  }, [props.appList, props.prefixCls]);
  var collapsedDom = useMemo(function () {
    if (collapsedButtonRender === false) return null;

    var dom = _jsx(CollapsedIcon, {
      isMobile: isMobile,
      collapsed: originCollapsed,
      className: "".concat(baseClassName, "-collapsed-button"),
      onClick: function onClick() {
        _onCollapse === null || _onCollapse === void 0 ? void 0 : _onCollapse(!originCollapsed);
      }
    });

    if (collapsedButtonRender) return collapsedButtonRender(collapsed, dom);
    return dom;
  }, [collapsedButtonRender, isMobile, originCollapsed, baseClassName, collapsed, _onCollapse]);
  var siderTitleViewCollapsedCss = useMemo(function () {
    return css(_templateObject9 || (_templateObject9 = _taggedTemplateLiteral(["\n      flex-direction: column-reverse;\n      .", "-basicLayout-apps-icon {\n        margin-bottom: 8px;\n      }\n    "])), prefixCls);
  }, [prefixCls]);
  /** 操作区域的dom */

  var actionAreaDom = useMemo(function () {
    if (!avatarDom && !actionsDom) return null;
    return _jsxs("div", {
      className: cx("".concat(baseClassName, "-actions"), collapsed && "".concat(baseClassName, "-actions-collapsed"), css(_templateObject10 || (_templateObject10 = _taggedTemplateLiteral(["\n            display: flex;\n            align-items: center;\n            justify-content: space-between;\n            margin: 4px 0;\n            padding: 0 16px;\n            color: ", ";\n          "])), designToken.sider.menuTextColor), collapsed && css(_templateObject11 || (_templateObject11 = _taggedTemplateLiteral(["\n              flex-direction: column-reverse;\n              padding: 0 8px;\n              font-size: 16px;\n              transition: font-size 0.3s;\n            "])))),
      children: [avatarDom, actionsDom]
    });
  }, [actionsDom, avatarDom, baseClassName, collapsed, designToken.sider.menuTextColor]);
  var antPrefix = context.getPrefixCls();
  var collapsedWidth = 60;
  /* Using the useMemo hook to create a CSS class that will hide the menu when the menu is collapsed. */

  var hideMenuWhenCollapsedCss = useMemo(function () {
    var _props$menu;

    // 收起时完全隐藏菜单
    if ((props === null || props === void 0 ? void 0 : (_props$menu = props.menu) === null || _props$menu === void 0 ? void 0 : _props$menu.hideMenuWhenCollapsed) && collapsed) {
      return css(_templateObject12 || (_templateObject12 = _taggedTemplateLiteral(["\n        left: -", "px;\n        position: absolute;\n      "])), collapsedWidth - 12);
    }

    return null;
  }, [collapsed, props === null || props === void 0 ? void 0 : (_props$menu2 = props.menu) === null || _props$menu2 === void 0 ? void 0 : _props$menu2.hideMenuWhenCollapsed]);
  var menuFooterDom = menuFooterRender && (menuFooterRender === null || menuFooterRender === void 0 ? void 0 : menuFooterRender(props));

  var menuDomItems = _jsxs(_Fragment, {
    children: [headerDom && _jsxs("div", {
      className: cx([classNames("".concat(baseClassName, "-logo"), _defineProperty({}, "".concat(baseClassName, "-logo-collapsed"), collapsed)), siderTitleViewCss === null || siderTitleViewCss === void 0 ? void 0 : siderTitleViewCss(designToken), collapsed && siderTitleViewCollapsedCss]),
      onClick: showSiderExtraDom ? onMenuHeaderClick : undefined,
      id: "logo",
      style: logoStyle,
      children: [headerDom, appsDom]
    }), extraDom && _jsx("div", {
      className: cx(["".concat(baseClassName, "-extra"), !headerDom && "".concat(baseClassName, "-extra-no-logo"), css(_templateObject13 || (_templateObject13 = _taggedTemplateLiteral(["\n              margin-bottom: 16px;\n              padding: 0 16px;\n            "]))), // no-logo
      !headerDom && css(_templateObject14 || (_templateObject14 = _taggedTemplateLiteral(["\n                margin-top: 16px;\n              "])))]),
      children: extraDom
    }), _jsx("div", {
      style: {
        flex: 1,
        overflowY: 'auto',
        overflowX: 'hidden'
      },
      children: menuRenderDom
    }), links ? _jsx("div", {
      className: cx("".concat(baseClassName, "-links"), css(_templateObject15 || (_templateObject15 = _taggedTemplateLiteral(["\n              width: 100%;\n              ul.", "-menu-root {\n                height: auto;\n              }\n            "])), antPrefix)),
      children: _jsx(_Menu, {
        inlineIndent: 16,
        className: cx("".concat(baseClassName, "-link-menu"), css({
          border: 'none',
          boxShadow: 'none',
          background: 'transparent'
        })),
        selectedKeys: [],
        openKeys: [],
        theme: "light",
        mode: "inline",
        items: linksMenuItems
      })
    }) : null, showSiderExtraDom && _jsxs(_Fragment, {
      children: [actionAreaDom, rightContentRender ? _jsx("div", {
        className: classNames("".concat(baseClassName, "-actions"), _defineProperty({}, "".concat(baseClassName, "-actions-collapsed"), collapsed)),
        children: rightContentRender === null || rightContentRender === void 0 ? void 0 : rightContentRender(props)
      }) : null]
    }), menuFooterDom && _jsx("div", {
      className: cx(["".concat(baseClassName, "-footer"), collapsed && "".concat(baseClassName, "-footer-collapsed"), css(_templateObject16 || (_templateObject16 = _taggedTemplateLiteral(["\n              color: ", ";\n              animation: ", " 0.35s;\n              padding-bottom: 16px;\n            "])), designToken.colorTextDisable, proLayoutTitleHide)]),
      children: menuFooterDom
    })]
  });

  useEffect(function () {
    return injectGlobal(_templateObject17 || (_templateObject17 = _taggedTemplateLiteral([" \n    .", "-menu-submenu > .", "-menu {\n      background-color: ", ";\n    }\n    "])), antPrefix, antPrefix, designToken.sider.menuBackgroundColor === 'transparent' ? '#fff' : designToken.sider.menuBackgroundColor);
  }, [antPrefix, designToken.sider.menuBackgroundColor]);
  return _jsxs(_Fragment, {
    children: [fixSiderbar && !isMobile && !hideMenuWhenCollapsedCss && _jsx("div", {
      className: cx(css(_objectSpread({
        width: collapsed ? collapsedWidth : siderWidth,
        overflow: 'hidden',
        flex: "0 0 ".concat(collapsed ? collapsedWidth : siderWidth, "px"),
        maxWidth: collapsed ? collapsedWidth : siderWidth,
        minWidth: collapsed ? collapsedWidth : siderWidth,
        transition: "background-color 0.3s, min-width 0.3s, max-width 0.3s cubic-bezier(0.645, 0.045, 0.355, 1)"
      }, style)))
    }), _jsxs(Sider, {
      collapsible: true,
      trigger: null,
      collapsed: collapsed,
      breakpoint: breakpoint === false ? undefined : breakpoint,
      onCollapse: function onCollapse(collapse) {
        if (isMobile) return;
        _onCollapse === null || _onCollapse === void 0 ? void 0 : _onCollapse(collapse);
      },
      collapsedWidth: collapsedWidth,
      style: _objectSpread({
        backgroundColor: designToken.sider.menuBackgroundColor
      }, style),
      width: siderWidth,
      theme: theme,
      className: cx(siderClassName, siderCss, css(_templateObject18 || (_templateObject18 = _taggedTemplateLiteral(["\n            .", "-layout-sider-children {\n              position: relative;\n              display: flex;\n              flex-direction: column;\n              height: 100%;\n              border-right: 1px solid ", ";\n            }\n          "])), antPrefix, designToken.borderColorSplit), fixSiderbar && css(_templateObject19 || (_templateObject19 = _taggedTemplateLiteral(["\n              position: fixed;\n              top: 0;\n              left: 0;\n              z-index: 100;\n              height: 100%;\n            "]))), hideMenuWhenCollapsedCss, layout === 'mix' && !isMobile && css(_templateObject20 || (_templateObject20 = _taggedTemplateLiteral(["\n              height: calc(100% - ", "px);\n              top: ", "px;\n            "])), headerHeight, headerHeight)),
      children: [hideMenuWhenCollapsedCss ? _jsx("div", {
        className: "".concat(baseClassName, "-hide-when-collapsed"),
        style: {
          height: '100%',
          width: '100%',
          opacity: hideMenuWhenCollapsedCss ? 0 : 1
        },
        children: menuDomItems
      }) : menuDomItems, collapsedDom]
    })]
  });
};

export { SiderMenu };
