// @ant-design/pro-layout/es/components/GlobalFooter/index.js
import _taggedTemplateLiteral from "@babel/runtime/helpers/esm/taggedTemplateLiteral";
import "antd/es/config-provider/style";
import _ConfigProvider from "antd/es/config-provider";

var _templateObject, _templateObject2, _templateObject3;

import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useContext } from 'react';
import { css, cx } from '../../emotion';
import { ProLayoutContext } from '../../ProLayoutContext';

var GlobalFooter = function GlobalFooter(_ref) {
  var className = _ref.className,
      prefixCls = _ref.prefixCls,
      links = _ref.links,
      copyright = _ref.copyright,
      style = _ref.style;
  var context = useContext(_ConfigProvider.ConfigContext);
  var designToken = useContext(ProLayoutContext);
  var baseClassName = context.getPrefixCls(prefixCls || 'pro-global-footer');

  if ((links == null || links === false || Array.isArray(links) && links.length === 0) && (copyright == null || copyright === false)) {
    return null;
  }

  return _jsxs("div", {
    className: cx(baseClassName, className, css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n          margin: 48px 0 24px 0;\n          padding: 0 16px;\n          text-align: center;\n        "])))),
    style: style,
    children: [links && _jsx("div", {
      className: cx("".concat(baseClassName, "-links"), css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n              margin-bottom: 8px;\n              a {\n                color: ", ";\n                transition: all 0.3s;\n                &:not(:last-child) {\n                  margin-right: 40px;\n                }\n                &:hover {\n                  color: ", ";\n                }\n              }\n            "])), designToken.colorTextSecondary, designToken.colorText)),
      children: links.map(function (link) {
        return _jsx("a", {
          title: link.key,
          target: link.blankTarget ? '_blank' : '_self',
          href: link.href,
          rel: "noreferrer",
          children: link.title
        }, link.key);
      })
    }), copyright && _jsx("div", {
      className: cx("".concat(baseClassName, "-copyright"), css(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n              font-size: 14px;\n              color: ", ";\n            "])), designToken.colorTextSecondary)),
      children: copyright
    })]
  });
};

export { GlobalFooter };
