// @ant-design/pro-layout/lib/Footer.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DefaultFooter = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

require("antd/lib/layout/style");

var _layout = _interopRequireDefault(require("antd/lib/layout"));

var _jsxRuntime = require("react/jsx-runtime");

var _icons = require("@ant-design/icons");

var _react = require("react");

var _GlobalFooter = require("./components/GlobalFooter");

var Footer = _layout.default.Footer;

var DefaultFooter = function DefaultFooter(_ref) {
  var links = _ref.links,
      copyright = _ref.copyright,
      style = _ref.style,
      className = _ref.className,
      prefixCls = _ref.prefixCls;
  return (0, _jsxRuntime.jsx)(Footer, {
    className: className,
    style: (0, _objectSpread2.default)({
      padding: 0
    }, style),
    children: (0, _jsxRuntime.jsx)(_GlobalFooter.GlobalFooter, {
      links: links,
      prefixCls: prefixCls,
      copyright: copyright === false ? null : (0, _jsxRuntime.jsxs)(_react.Fragment, {
        children: [(0, _jsxRuntime.jsx)(_icons.CopyrightOutlined, {}), " ", copyright]
      })
    })
  });
};

exports.DefaultFooter = DefaultFooter;
