// @ant-design/pro-layout/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "DefaultFooter", {
  enumerable: true,
  get: function get() {
    return _Footer.DefaultFooter;
  }
});
Object.defineProperty(exports, "DefaultHeader", {
  enumerable: true,
  get: function get() {
    return _Header.DefaultHeader;
  }
});
Object.defineProperty(exports, "FooterToolbar", {
  enumerable: true,
  get: function get() {
    return _FooterToolbar.FooterToolbar;
  }
});
Object.defineProperty(exports, "GridContent", {
  enumerable: true,
  get: function get() {
    return _GridContent.GridContent;
  }
});
Object.defineProperty(exports, "PageContainer", {
  enumerable: true,
  get: function get() {
    return _PageContainer.PageContainer;
  }
});
Object.defineProperty(exports, "PageLoading", {
  enumerable: true,
  get: function get() {
    return _PageLoading.PageLoading;
  }
});
Object.defineProperty(exports, "ProBreadcrumb", {
  enumerable: true,
  get: function get() {
    return _PageContainer.ProBreadcrumb;
  }
});
Object.defineProperty(exports, "ProLayout", {
  enumerable: true,
  get: function get() {
    return _ProLayout.ProLayout;
  }
});
Object.defineProperty(exports, "ProPageHeader", {
  enumerable: true,
  get: function get() {
    return _PageContainer.ProPageHeader;
  }
});
Object.defineProperty(exports, "RouteContext", {
  enumerable: true,
  get: function get() {
    return _RouteContext.RouteContext;
  }
});
Object.defineProperty(exports, "SettingDrawer", {
  enumerable: true,
  get: function get() {
    return _SettingDrawer.SettingDrawer;
  }
});
Object.defineProperty(exports, "TopNavHeader", {
  enumerable: true,
  get: function get() {
    return _TopNavHeader.TopNavHeader;
  }
});
Object.defineProperty(exports, "WaterMark", {
  enumerable: true,
  get: function get() {
    return _WaterMark.WaterMark;
  }
});
exports.default = void 0;
Object.defineProperty(exports, "getMenuData", {
  enumerable: true,
  get: function get() {
    return _getMenuData.getMenuData;
  }
});
Object.defineProperty(exports, "getPageTitle", {
  enumerable: true,
  get: function get() {
    return _getPageTitle.getPageTitle;
  }
});

var _FooterToolbar = require("./components/FooterToolbar");

var _GridContent = require("./components/GridContent");

var _PageContainer = require("./components/PageContainer");

var _PageLoading = require("./components/PageLoading");

var _SettingDrawer = require("./components/SettingDrawer");

var _TopNavHeader = require("./components/TopNavHeader");

var _WaterMark = require("./components/WaterMark");

var _Footer = require("./Footer");

var _getPageTitle = require("./getPageTitle");

var _Header = require("./Header");

var _ProLayout = require("./ProLayout");

var _RouteContext = require("./RouteContext");

var _getMenuData = require("./utils/getMenuData");

var _default = _ProLayout.ProLayout;
exports.default = _default;
