// @ant-design/pro-layout/es/ProLayoutContext.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread2";
import { jsx as _jsx } from "react/jsx-runtime";
import React from 'react';
export var baseDesignTokens = {
  /**
   * 主色
   */
  primaryColor: '#1677FF',

  /**
   *错误色
   */
  errorColor: '#ff4d4f',

  /**
   * 成功色
   */
  successColor: '#1677FF',

  /**
   *通知色
   */
  infoColor: '#1677FF',

  /**
   *警告色
   */
  warningColor: '#faad14',

  /**
   * 正文色
   */
  colorText: 'rgba(0, 0, 0, 0.65)',

  /**
   * 标题色
   */
  colorHeading: 'rgba(0, 0, 0, 0.85)',

  /**
   * 次级色
   */
  colorTextSecondary: 'rgba(0, 0, 0, 0.45)',

  /**
   * 禁用的文本颜色
   */
  colorTextDisable: 'rgba(0, 0, 0, 0.25)',

  /**
   * 边框基本色
   */
  borderColorBase: '#d9d9d9',

  /**
   * 分隔边框的颜色
   */
  borderColorSplit: 'rgba(0, 0, 0, 0.06)',

  /**
   * 默认阴影
   */
  boxShadowBase: '',

  /**
   * 默认圆角
   */
  borderRadiusBase: '4px'
}; // DesignToken 的默认样式

export var DefaultDesignToken = {
  layoutBgColor: '#f7f8fa',
  layoutBg: 'linear-gradient(#fff, #f7f8fa 28%)',
  appListIconTextColor: '#666',
  menuToken: {
    menuItemHoverBgColor: 'rgba(90, 75, 75, 0.03)',
    menuItemSelectedBgColor: 'rgba(0, 0, 0, 0.04)',
    menuItemCollapsedHoverBgColor: 'rgba(90, 75, 75, 0.03)',
    menuItemCollapsedSelectedBgColor: 'rgba(0, 0, 0, 0.04)',
    menuSelectedTextColor: 'rgb(0,0,0,0.95)',
    subMenuSelectedTextColor: 'rgb(0,0,0,0.95)',
    horizontalMenuItemPadding: '0 20px'
  },
  header: {
    headerBgColor: 'rgba(240, 242, 245, 0.4)',
    rightActionsItemHoverBgColor: 'rgba(0, 0, 0, 0.03)'
  },
  sider: {
    menuBackgroundColor: 'transparent',
    menuSubArrowColor: 'rgba(0, 0, 0, 0.25)',
    collapsedButtonBgColor: '#fff'
  },
  pageContainer: {
    pageContainerBgColor: 'transparent',
    pageContainerFixedBgColor: '#fff',
    pageContainerContentMargin: '0px 40px 0px 40px'
  }
};
export var getLayoutDesignToken = function getLayoutDesignToken(designTokens) {
  var finalDesignTokens = _objectSpread(_objectSpread({}, baseDesignTokens), designTokens);

  var menuToken = _objectSpread({
    menuTextColor: finalDesignTokens.colorText,
    menuTextColorSecondary: finalDesignTokens.colorTextSecondary
  }, DefaultDesignToken.menuToken);

  return _objectSpread(_objectSpread(_objectSpread({}, designTokens), {}, {
    layoutBgColor: '#f7f8fa',
    layoutBg: 'linear-gradient(#fff, #f7f8fa 28%)',
    appListIconTextColor: '#666',
    appListIconHoverTextColor: finalDesignTokens.colorText,
    appListIconHoverBgColor: menuToken.menuItemSelectedBgColor
  }, finalDesignTokens), {}, {
    header: _objectSpread(_objectSpread(_objectSpread({}, menuToken), {}, {
      headerTitleColor: finalDesignTokens.colorHeading,
      rightActionsItemTextColor: finalDesignTokens.colorTextSecondary
    }, DefaultDesignToken.header), finalDesignTokens.header),
    sider: _objectSpread(_objectSpread(_objectSpread({}, menuToken), {}, {
      menuItemDividerColor: finalDesignTokens.borderColorSplit,
      menuTitleTextColor: finalDesignTokens.colorHeading,
      collapsedButtonTextColor: menuToken === null || menuToken === void 0 ? void 0 : menuToken.menuTextColorSecondary,
      collapsedButtonHoverTextColor: menuToken === null || menuToken === void 0 ? void 0 : menuToken.menuTextColor
    }, DefaultDesignToken.sider), finalDesignTokens.sider),
    pageContainer: _objectSpread(_objectSpread({}, DefaultDesignToken.pageContainer), finalDesignTokens.pageContainer)
  });
};
var defaultToken = getLayoutDesignToken(baseDesignTokens);
export var ProLayoutContext = /*#__PURE__*/React.createContext(defaultToken);
export var ProLayoutProvider = function ProLayoutProvider(props) {
  return _jsx(ProLayoutContext.Provider, {
    value: _objectSpread({}, getLayoutDesignToken((props === null || props === void 0 ? void 0 : props.token) || {})),
    children: props.children
  });
};
