// @ant-design/pro-layout/es/components/AppsLogoComponents/DefaultContent.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useContext } from 'react';
import { cx } from '../../emotion';
import { ProLayoutContext } from '../../ProLayoutContext';
import { appContentListCss, getAppContentLisItem } from './css';
import { defaultRenderLogo } from './index';
export var DefaultContent = function DefaultContent(props) {
  var designToken = useContext(ProLayoutContext);
  var appList = props.appList,
      _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'ant' : _props$prefixCls;
  return _jsx("div", {
    className: "".concat(prefixCls, "-basicLayout-apps-content"),
    children: _jsx("ul", {
      className: cx("".concat(prefixCls, "-basicLayout-apps-content-list"), appContentListCss),
      children: appList === null || appList === void 0 ? void 0 : appList.map(function (app, index) {
        return _jsx("li", {
          className: cx("".concat(prefixCls, "-basicLayout-apps-content-list-item"), getAppContentLisItem(designToken)),
          children: _jsxs("a", {
            href: app.url,
            target: app.target,
            rel: "noreferrer",
            children: [defaultRenderLogo(app.icon), _jsxs("div", {
              children: [_jsx("div", {
                children: app.title
              }), app.desc ? _jsx("span", {
                children: app.desc
              }) : null]
            })]
          })
        }, index);
      })
    })
  });
};
