// @ant-design/pro-layout/es/components/AppsLogoComponents/index.js
import "antd/es/popover/style";
import _Popover from "antd/es/popover";
import "antd/es/config-provider/style";
import _ConfigProvider from "antd/es/config-provider";
import { jsx as _jsx } from "react/jsx-runtime";
import { useContext, useMemo } from 'react';
import { cx } from '../../emotion';
import { ProLayoutContext } from '../../ProLayoutContext';
import { AppsLogo } from './AppsLogo';
import { appIconsCss, getAntdPopoverContentListCss } from './css';
import { DefaultContent } from './DefaultContent';
import { SimpleContent } from './SimpleContent';
/**
 * 默认渲染logo的方式，如果是个string，用img。否则直接返回
 *
 * @param logo
 * @returns
 */

export var defaultRenderLogo = function defaultRenderLogo(logo) {
  if (typeof logo === 'string') {
    return _jsx("img", {
      src: logo,
      alt: "logo"
    });
  }

  if (typeof logo === 'function') {
    return logo();
  }

  return logo;
};
/**
 * 相关品牌额icon 列表。用于展示相关的品牌
 *
 * @param props
 * @returns
 */

export var AppsLogoComponents = function AppsLogoComponents(props) {
  var _props$appList;

  var designToken = useContext(ProLayoutContext);
  var appList = props.appList,
      _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'ant' : _props$prefixCls;
  var antdContext = useContext(_ConfigProvider.ConfigContext);
  var antdPreFixCls = antdContext.getPrefixCls();
  var popoverContent = useMemo(function () {
    var isSimple = appList === null || appList === void 0 ? void 0 : appList.some(function (app) {
      return !(app === null || app === void 0 ? void 0 : app.desc);
    });

    if (isSimple) {
      return _jsx(SimpleContent, {
        appList: appList,
        prefixCls: prefixCls
      });
    }

    return _jsx(DefaultContent, {
      appList: appList,
      prefixCls: prefixCls
    });
  }, [appList, prefixCls]);
  if (!(props === null || props === void 0 ? void 0 : (_props$appList = props.appList) === null || _props$appList === void 0 ? void 0 : _props$appList.length)) return null;
  return _jsx(_Popover, {
    placement: "bottomRight",
    trigger: ['click'],
    zIndex: 9999,
    arrowPointAtCenter: true,
    overlayClassName: cx(getAntdPopoverContentListCss(antdPreFixCls)),
    content: popoverContent,
    children: _jsx("span", {
      onClick: function onClick(e) {
        e.stopPropagation();
      },
      className: cx("".concat(prefixCls, "-basicLayout-apps-icon"), appIconsCss(designToken)),
      children: _jsx(AppsLogo, {})
    })
  });
};
