// @ant-design/pro-layout/es/components/AppsLogoComponents/SimpleContent.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useContext } from 'react';
import { cx } from '../../emotion';
import { ProLayoutContext } from '../../ProLayoutContext';
import { appContentListCss, getAppContentLisItem } from './simple';
/**
 * simple模式渲染logo的方式
 *
 * @param logo
 * @param title
 * @returns
 */

export var renderLogo = function renderLogo(logo, title) {
  if (logo && typeof logo === 'string') {
    return _jsx("img", {
      src: logo,
      alt: "logo"
    });
  }

  if (typeof logo === 'function') {
    return logo();
  }

  if (!logo && title && typeof title === 'string') {
    var symbol = title.substring(0, 1);
    return _jsx("div", {
      id: "avatarLogo",
      children: symbol
    });
  }

  return logo;
};
export var SimpleContent = function SimpleContent(props) {
  var designToken = useContext(ProLayoutContext);
  var appList = props.appList,
      _props$prefixCls = props.prefixCls,
      prefixCls = _props$prefixCls === void 0 ? 'ant' : _props$prefixCls;
  return _jsx("div", {
    className: "".concat(prefixCls, "-basicLayout-apps-content"),
    children: _jsx("ul", {
      className: cx("".concat(prefixCls, "-basicLayout-apps-content-list"), appContentListCss),
      children: appList === null || appList === void 0 ? void 0 : appList.map(function (app, index) {
        return _jsx("li", {
          className: cx("".concat(prefixCls, "-basicLayout-apps-content-list-item"), getAppContentLisItem(designToken)),
          children: _jsxs("a", {
            href: app.url,
            target: app.target,
            rel: "noreferrer",
            children: [renderLogo(app.icon, app.title), _jsx("div", {
              children: _jsx("div", {
                children: app.title
              })
            })]
          })
        }, index);
      })
    })
  });
};
