// @ant-design/pro-layout/lib/components/GlobalFooter/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GlobalFooter = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

require("antd/lib/config-provider/style");

var _configProvider = _interopRequireDefault(require("antd/lib/config-provider"));

var _jsxRuntime = require("react/jsx-runtime");

var _react = require("react");

var _emotion = require("../../emotion");

var _ProLayoutContext = require("../../ProLayoutContext");

var _templateObject, _templateObject2, _templateObject3;

var GlobalFooter = function GlobalFooter(_ref) {
  var className = _ref.className,
      prefixCls = _ref.prefixCls,
      links = _ref.links,
      copyright = _ref.copyright,
      style = _ref.style;
  var context = (0, _react.useContext)(_configProvider.default.ConfigContext);
  var designToken = (0, _react.useContext)(_ProLayoutContext.ProLayoutContext);
  var baseClassName = context.getPrefixCls(prefixCls || 'pro-global-footer');

  if ((links == null || links === false || Array.isArray(links) && links.length === 0) && (copyright == null || copyright === false)) {
    return null;
  }

  return (0, _jsxRuntime.jsxs)("div", {
    className: (0, _emotion.cx)(baseClassName, className, (0, _emotion.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2.default)(["\n          margin: 48px 0 24px 0;\n          padding: 0 16px;\n          text-align: center;\n        "])))),
    style: style,
    children: [links && (0, _jsxRuntime.jsx)("div", {
      className: (0, _emotion.cx)("".concat(baseClassName, "-links"), (0, _emotion.css)(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2.default)(["\n              margin-bottom: 8px;\n              a {\n                color: ", ";\n                transition: all 0.3s;\n                &:not(:last-child) {\n                  margin-right: 40px;\n                }\n                &:hover {\n                  color: ", ";\n                }\n              }\n            "])), designToken.colorTextSecondary, designToken.colorText)),
      children: links.map(function (link) {
        return (0, _jsxRuntime.jsx)("a", {
          title: link.key,
          target: link.blankTarget ? '_blank' : '_self',
          href: link.href,
          rel: "noreferrer",
          children: link.title
        }, link.key);
      })
    }), copyright && (0, _jsxRuntime.jsx)("div", {
      className: (0, _emotion.cx)("".concat(baseClassName, "-copyright"), (0, _emotion.css)(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2.default)(["\n              font-size: 14px;\n              color: ", ";\n            "])), designToken.colorTextSecondary)),
      children: copyright
    })]
  });
};

exports.GlobalFooter = GlobalFooter;
