// @ant-design/pro-layout/lib/components/TopNavHeader/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TopNavHeader = exports.RightContent = void 0;

require("antd/lib/config-provider/style");

var _configProvider = _interopRequireDefault(require("antd/lib/config-provider"));

var _regeneratorRuntime2 = _interopRequireDefault(require("@babel/runtime/helpers/regeneratorRuntime"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));

require("antd/lib/avatar/style");

var _avatar = _interopRequireDefault(require("antd/lib/avatar"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _jsxRuntime = require("react/jsx-runtime");

var _react = _interopRequireWildcard(require("react"));

var _proUtils = require("@ant-design/pro-utils");

var _rcResizeObserver = _interopRequireDefault(require("rc-resize-observer"));

var _emotion = require("../../emotion");

var _ProLayoutContext = require("../../ProLayoutContext");

var _AppsLogoComponents = require("../AppsLogoComponents");

var _BaseMenu = require("../SiderMenu/BaseMenu");

var _SiderMenu = require("../SiderMenu/SiderMenu");

var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6, _templateObject7, _templateObject8, _templateObject9, _templateObject10;

var _excluded = ["rightContentRender", "prefixCls", "avatarProps", "actionsRender", "headerContentRender"],
    _excluded2 = ["title"];

/**
 * 抽离出来是为了防止 rightSize 经常改变导致菜单 render
 *
 * @param param0
 */
var RightContent = function RightContent(_ref) {
  var rightContentRender = _ref.rightContentRender,
      prefixCls = _ref.prefixCls,
      avatarProps = _ref.avatarProps,
      actionsRender = _ref.actionsRender,
      headerContentRender = _ref.headerContentRender,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var designToken = (0, _react.useContext)(_ProLayoutContext.ProLayoutContext);

  var _useState = (0, _react.useState)('auto'),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      rightSize = _useState2[0],
      setRightSize = _useState2[1];

  var avatarDom = (0, _react.useMemo)(function () {
    if (!avatarProps) return null;
    var title = avatarProps.title,
        rest = (0, _objectWithoutProperties2.default)(avatarProps, _excluded2);
    return [/*#__PURE__*/(0, _react.createElement)(_avatar.default, (0, _objectSpread2.default)((0, _objectSpread2.default)({
      size: "large"
    }, rest), {}, {
      key: "avatar"
    })), title ? (0, _jsxRuntime.jsx)("span", {
      style: {
        marginLeft: 8
      },
      children: title
    }, "name") : undefined];
  }, [avatarProps]);

  var rightActionsRender = function rightActionsRender(restParams) {
    var doms = actionsRender && (actionsRender === null || actionsRender === void 0 ? void 0 : actionsRender(restParams));
    if (!doms && !avatarDom) return null;
    if (!Array.isArray(doms)) doms = [doms];
    return (0, _jsxRuntime.jsxs)("div", {
      className: (0, _emotion.cx)("".concat(prefixCls, "-header-actions"), (0, _emotion.css)(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2.default)(["\n            display: flex;\n            height: 100%;\n          "])))),
      children: [doms.filter(Boolean).map(function (dom, index) {
        var _designToken$header, _designToken$header2;

        var hideHover = false; // 如果配置了 hideHover 就不展示 hover 效果了

        if ( /*#__PURE__*/_react.default.isValidElement(dom)) {
          var _dom$props;

          hideHover = !!(dom === null || dom === void 0 ? void 0 : (_dom$props = dom.props) === null || _dom$props === void 0 ? void 0 : _dom$props['aria-hidden']);
        }

        return (0, _jsxRuntime.jsx)("div", {
          className: (0, _emotion.cx)("".concat(prefixCls, "-header-actions-item"), (0, _emotion.css)(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2.default)(["\n                  display: inline-flex;\n                  align-items: center;\n                  justify-content: center;\n                  padding: 0px 6px;\n                  font-size: 16px;\n                  color: ", ";\n                  cursor: pointer;\n                  border-radius: ", ";\n                "])), designToken === null || designToken === void 0 ? void 0 : (_designToken$header = designToken.header) === null || _designToken$header === void 0 ? void 0 : _designToken$header.rightActionsItemTextColor, designToken.borderRadiusBase), !hideHover && (0, _emotion.css)(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2.default)(["\n                    > * {\n                      padding: 6px;\n                      border-radius: ", ";\n                      &:hover {\n                        background-color: ", ";\n                      }\n                    }\n                  "])), designToken.borderRadiusBase, (_designToken$header2 = designToken.header) === null || _designToken$header2 === void 0 ? void 0 : _designToken$header2.rightActionsItemHoverBgColor)),
          children: dom
        }, index);
      }), avatarDom && (0, _jsxRuntime.jsx)("span", {
        className: (0, _emotion.cx)("".concat(prefixCls, "-header-actions-item"), (0, _emotion.css)(_templateObject4 || (_templateObject4 = (0, _taggedTemplateLiteral2.default)(["\n                display: inline-flex;\n                align-items: center;\n                justify-content: center;\n                padding-left: 16px;\n                padding-right: 16px;\n              "])))),
        children: (0, _jsxRuntime.jsx)("div", {
          className: (0, _emotion.css)(_templateObject5 || (_templateObject5 = (0, _taggedTemplateLiteral2.default)(["\n                height: 44px;\n                padding: 8px;\n                cursor: pointer;\n                display: flex;\n                align-items: center;\n                line-height: 44px;\n                border-radius: ", ";\n                &:hover {\n                  background-color: rgba(0, 0, 0, 0.03);\n                }\n              "])), designToken.borderRadiusBase),
          children: avatarDom
        })
      })]
    });
  };
  /** 减少一下渲染的次数 */


  var setRightSizeDebounceFn = (0, _proUtils.useDebounceFn)( /*#__PURE__*/function () {
    var _ref2 = (0, _asyncToGenerator2.default)( /*#__PURE__*/(0, _regeneratorRuntime2.default)().mark(function _callee(width) {
      return (0, _regeneratorRuntime2.default)().wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              setRightSize(width);

            case 1:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));

    return function (_x) {
      return _ref2.apply(this, arguments);
    };
  }(), 160);
  return (0, _jsxRuntime.jsx)("div", {
    className: "".concat(prefixCls, "-right-content"),
    style: {
      minWidth: rightSize,
      height: '100%'
    },
    children: (0, _jsxRuntime.jsx)("div", {
      style: {
        height: '100%'
      },
      children: (0, _jsxRuntime.jsx)(_rcResizeObserver.default, {
        onResize: function onResize(_ref3) {
          var width = _ref3.width;
          setRightSizeDebounceFn.run(width);
        },
        children: (rightContentRender || rightActionsRender) && (0, _jsxRuntime.jsx)("div", {
          style: {
            display: 'flex',
            alignItems: 'center',
            height: '100%',
            justifyContent: 'flex-end'
          },
          children: (rightContentRender || rightActionsRender)((0, _objectSpread2.default)((0, _objectSpread2.default)({}, props), {}, {
            // 测试专用
            //@ts-ignore
            rightContentSize: rightSize
          }))
        })
      })
    })
  });
};

exports.RightContent = RightContent;

var TopNavHeader = function TopNavHeader(props) {
  var _designToken$header3;

  var ref = (0, _react.useRef)(null);
  var onMenuHeaderClick = props.onMenuHeaderClick,
      contentWidth = props.contentWidth,
      rightContentRender = props.rightContentRender,
      propsClassName = props.className,
      style = props.style,
      headerContentRender = props.headerContentRender,
      layout = props.layout,
      actionsRender = props.actionsRender;
  var designToken = (0, _react.useContext)(_ProLayoutContext.ProLayoutContext);

  var _useContext = (0, _react.useContext)(_configProvider.default.ConfigContext),
      getPrefixCls = _useContext.getPrefixCls;

  var antdPreFix = getPrefixCls();
  var prefixCls = "".concat(props.prefixCls || getPrefixCls('pro'), "-top-nav-header");
  var headerDom = (0, _SiderMenu.renderLogoAndTitle)((0, _objectSpread2.default)((0, _objectSpread2.default)({}, props), {}, {
    collapsed: false
  }), layout === 'mix' ? 'headerTitleRender' : undefined);
  var contentDom = (0, _react.useMemo)(function () {
    var defaultDom = (0, _jsxRuntime.jsx)(_BaseMenu.BaseMenu, (0, _objectSpread2.default)((0, _objectSpread2.default)((0, _objectSpread2.default)({
      theme: "light"
    }, props), props.menuProps), {}, {
      collapsed: false,
      menuRenderType: "header",
      mode: "horizontal"
    }));

    if (headerContentRender) {
      return headerContentRender(props, defaultDom);
    }

    return defaultDom;
  }, [headerContentRender, props]);
  return (0, _jsxRuntime.jsx)("div", {
    className: (0, _emotion.cx)(prefixCls, propsClassName, {
      light: true
    }, (0, _emotion.css)(_templateObject6 || (_templateObject6 = (0, _taggedTemplateLiteral2.default)(["\n          position: relative;\n          width: 100%;\n          height: 100%;\n          background-color: transparent;\n\n          .", "-menu {\n            background: transparent;\n          }\n          .anticon {\n            color: inherit;\n          }\n        "])), antdPreFix)),
    style: style,
    children: (0, _jsxRuntime.jsxs)("div", {
      ref: ref,
      className: (0, _emotion.cx)("".concat(prefixCls, "-main"), "".concat(contentWidth === 'Fixed' ? 'wide' : ''), (0, _emotion.css)(_templateObject7 || (_templateObject7 = (0, _taggedTemplateLiteral2.default)(["\n            display: flex;\n            height: 100%;\n            padding-left: 16px;\n          "])))),
      children: [headerDom && (0, _jsxRuntime.jsxs)("div", {
        className: (0, _emotion.cx)("".concat(prefixCls, "-main-left"), (0, _emotion.css)(_templateObject8 || (_templateObject8 = (0, _taggedTemplateLiteral2.default)(["\n                display: flex;\n                align-items: center;\n                min-width: 192px;\n                .", "-pro-basicLayout-apps-icon {\n                  margin-right: 16px;\n                }\n              "])), antdPreFix)),
        onClick: onMenuHeaderClick,
        children: [(0, _jsxRuntime.jsx)(_AppsLogoComponents.AppsLogoComponents, (0, _objectSpread2.default)({}, props)), (0, _jsxRuntime.jsx)("div", {
          className: (0, _emotion.cx)("".concat(prefixCls, "-logo"), (0, _emotion.css)(_templateObject9 || (_templateObject9 = (0, _taggedTemplateLiteral2.default)(["\n                  position: relative;\n                  min-width: 165px;\n                  display: flex;\n                  height: 100%;\n                  overflow: hidden;\n                  a {\n                    display: flex;\n                    align-items: center;\n                    min-height: 22px;\n                    font-size: 22px;\n                  }\n                  img {\n                    display: inline-block;\n                    height: 32px;\n                    vertical-align: middle;\n                  }\n\n                  h1 {\n                    display: inline-block;\n                    margin: 0 0 0 6px;\n                    color: ", ";\n                    font-weight: 600;\n                    font-size: 16px;\n                    vertical-align: top;\n                  }\n                "])), designToken === null || designToken === void 0 ? void 0 : (_designToken$header3 = designToken.header) === null || _designToken$header3 === void 0 ? void 0 : _designToken$header3.headerTitleColor)),
          id: "logo",
          children: headerDom
        }, "logo")]
      }), (0, _jsxRuntime.jsx)("div", {
        style: {
          flex: 1
        },
        className: (0, _emotion.cx)("".concat(prefixCls, "-menu"), (0, _emotion.css)(_templateObject10 || (_templateObject10 = (0, _taggedTemplateLiteral2.default)(["\n              min-width: 0;\n              display: flex;\n              align-items: center;\n              .", "-menu.", "-menu-horizontal {\n                height: 100%;\n                border: none;\n              }\n            "])), antdPreFix, antdPreFix)),
        children: contentDom
      }), (rightContentRender || actionsRender || props.avatarProps) && (0, _jsxRuntime.jsx)(RightContent, (0, _objectSpread2.default)({
        rightContentRender: rightContentRender
      }, props))]
    })
  });
};

exports.TopNavHeader = TopNavHeader;
